### PR TITLE
(fix) O3-3749: Ward Allocation - Create new bed form missing scroll bar.

### PR DESCRIPTION
**Pull request by IamMujuziMoses**
## Requirements

- [x] This PR has a title that briefly describes the work done including the ticket number. If there is a ticket, make sure your PR title includes a [conventional commit](https://o3-docs.openmrs.org/docs/frontend-modules/contributing.en-US#contributing-guidelines) label. See existing PR titles for inspiration.
- [x] My work conforms to the [OpenMRS 3.0 Styleguide](https://om.rs/styleguide) and [design documentation](https://om.rs/o3ui).
- [ ] My work includes tests or is validated by existing tests.

## Summary

The Create a new bed form is currently missing a scroll bar, this PR adds a Scrollbar to the dialog to make it easy for different screen sizes.

## Screenshots

https://github.com/user-attachments/assets/7beacf7b-ac19-4b36-8c44-7db64d962a99

## Related Issue

https://issues.openmrs.org/browse/O3-3749

### CODE REVIEW

**Comment by IamMujuziMoses** [comment]
@ibacher @samuelmale could you please review this PR, your feedback will be appreciated.

File: packages/esm-bed-management-app/src/bed-administration/bed-administration-form.component.tsx [hunk 1]
```diff
@@ -119,7 +120,7 @@ const BedAdministrationForm: React.FC<BedAdministrationFormProps> = ({
     <ComposedModal open={showModal} onClose={() => onModalChange(false)} preventCloseOnClickOutside>
       <ModalHeader title={headerTitle} />
       <Form onSubmit={handleSubmit(onSubmit, onError)}>
-        <ModalBody hasScrollingContent>
```
Comment (vasharma05) [inline packages/esm-bed-management-app/src/bed-administration/bed-administration-form.component.tsx:122]: Hi @IamMujuziMoses @chibongho, 
We should also pass `size` along with the `hasScrollingContent` prop for the scrolling to work, please test it out.
Thanks!

**Comment by chibongho** [comment]
I have done similar work to fix this problem [elsewhere](https://openmrs.atlassian.net/browse/O3-3188?focusedCommentId=140847). 

This is not documented in Carbon IFAIK, but this can be fixed without css by making sure that the `<ModalHeader>`, `<ModalBody>` and `<ModalFooter>` are direct children of `ContentModal`. I tested locally that rearranging the elements like this seems to work:

```
<Form>
  <ContentModal>
    <ModalHeader/>
    <ModalBody/>
    <ModalFooter/>
  </ContentModal>
</Form>
```

There are a few other instances of `ComposedModal` usage in the bed management module seem to have this problem as well. @IamMujuziMoses , do you want to try to fix those too?

**Comment by IamMujuziMoses** [comment]
> but this can be fixed without css by making sure that the `<ModalHeader>`, `<ModalBody>` and `<ModalFooter>` are direct children of `ContentModal`. I tested locally that rearranging the elements like this seems to work:
> 
> ```
> <Form>
>   <ContentModal>
>     <ModalHeader/>
>     <ModalBody/>
>     <ModalFooter/>
>   </ContentModal>
> </Form>
> ```
> 

@chibongho This could work but there is a white line that appears, don't know what brings it, shown in the video below:

https://github.com/user-attachments/assets/192af5d9-cadc-496c-a3c0-14dbbc8a8c2d

**Comment by IamMujuziMoses** [comment]
> There are a few other instances of `ComposedModal` usage in the bed management module seem to have this problem as well. @IamMujuziMoses , do you want to try to fix those too?

yes, let me get the best solution for this first then I will make a ticket on Jira🙂

**Comment by IamMujuziMoses** [comment]
@chibongho @vasharma05, can you review this update, this is how it appears now:

https://github.com/user-attachments/assets/1236b19f-c9c2-4868-a49d-679cfcc25980

File: packages/esm-bed-management-app/src/bed-administration/bed-administration-form.component.tsx [hunk 2]
```diff
@@ -275,16 +275,16 @@ const BedAdministrationForm: React.FC<BedAdministrationFormProps> = ({
...
-          <Button disabled={!isDirty} type="submit">
-            <span>{t('save', 'Save')}</span>
-          </Button>
-        </ModalFooter>
-      </Form>
+        </Form>
+      </ModalBody>
+      <ModalFooter>
+        <Button onClick={() => onModalChange(false)} kind="secondary">
+          {t('cancel', 'Cancel')}
```
Comment (usamaidrsk) [inline packages/esm-bed-management-app/src/bed-administration/bed-administration-form.component.tsx:282]: Hi @IamMujuziMoses , you could make use of the framework `getCoreTranslation` 
Thanks

**Review by usamaidrsk — COMMENTED** [review]
Hi @IamMujuziMoses , this looks good.
I do have a small suggestion though. 
Thanks.

**Review by usamaidrsk — APPROVED** [review]

**Review by chibongho — APPROVED** [review]
Thanks @IamMujuziMoses . Looks great. Bummer that you had to change the footer button (albeit slightly) to get this to work.

**Pull request merged**